capa/features/extractors/ts/engine.py

```diff
@@ -0,0 +1,114 @@
+            yield self.get_object_id(obj_node)
+
+    def get_import_names(self, node: Node) -> List[Tuple[Node, str]]:
+        join_names = capa.features.extractors.ts.sig.get_name_joiner(self.language)
+        import_names = []
+        namespaces = set([self.get_range(node) for node, _ in self.get_all_namespaces()])
+        for node, _ in self.get_new_object_ids(node):
+            for namespace in namespaces:
+                name = join_names(namespace, self.get_range(node))
+                if name in self.import_signatures:
+                    import_names.append(name)
+        return import_names
+
+    def get_function_definitions(self, node: Node = None) -> List[Tuple[Node, str]]:
+        return self.query.function_definition.captures(node if node is not None else self.tree.root_node)
+
+    def get_function_definition_id(self, node: Node) -> Node:
+        return node.child_by_field_name(self.query.function_definition_field_name)
+
+    def get_function_definition_ids(self, node: Node) -> Iterator[Node]:
+        for fn_node, _ in self.get_function_definitions(node):
+            yield self.get_function_definition_id(fn_node)
+
+    def get_function_calls(self, node: Node) -> List[Tuple[Node, str]]:
+        return self.query.function_call.captures(node)
+
+    def get_function_call_id(self, node: Node) -> Node:
+        return node.child_by_field_name(self.query.function_call_field_name)
+
+    def get_function_call_ids(self, node: Node) -> Iterator[Node]:
+        for fn_node, _ in self.get_function_calls(node):
+            yield self.get_function_call_id(fn_node)
+
+    def get_function_names(self, node: Node) -> List[Tuple[Node, str]]:
+        join_names = capa.features.extractors.ts.sig.get_name_joiner(self.language)
+        function_names = []
+        namespaces = set([self.get_range(node) for node, _ in self.get_all_namespaces()])
+        for node, _ in self.get_function_call_ids(node):
+            for namespace in namespaces:
+                name = join_names(namespace, self.get_range(node))
+                if name in self.import_signatures:
+                    function_names.append(name)
+        return function_names
+
+    def get_string_literals(self, node: Node) -> List[Tuple[Node, str]]:
+        return self.query.string_literal.captures(node)
+
+    def get_integer_literals(self, node: Node) -> List[Tuple[Node, str]]:
+        return self.query.integer_literal.captures(node)
+
+    def get_namespaces(self, node: Node) -> List[Tuple[Node, str]]:
+        return self.query.namespace.captures(node)
+
+    def get_all_namespaces(self) -> List[Tuple[Node, str]]:
+        return self.get_namespaces(self.tree.root_node)
+
+    def get_global_statements(self) -> List[Tuple[Node, str]]:
+        return self.query.global_statement.captures(self.tree.root_node)
+
+    def get_range(self, node: Node) -> str:
+        return self.buf[node.start_byte : node.end_byte].decode()
```

Comment:
we should specify an encoding here to be extra explicit (and make it easier to grep for "utf-8" when triaging bugs)